### PR TITLE
Don't do EFI stuff if on unsupported architecture

### PR DESCRIPTION
**Pull request by DaanDeMeyer**

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
Pushed a bunch of other commits into this as well.

File: mkosi/__init__.py [hunk 1]
```diff
@@ -998,11 +1001,14 @@ def build_uki(
 
     cmdline += state.config.kernel_command_line
 
+    if not (arch := state.config.architecture.to_efi()):
+        die(f"Architecture {state.config.architecture} does not support UEFI")
+
```
Comment (behrmann) [inline mkosi/__init__.py:1006]: Is there a need to have this check above the comment? Having it closer to the use in the definition of `stub` would make it easier to follow where `arch` comes from there.

File: mkosi/__init__.py [hunk 2]
```diff
@@ -1675,7 +1684,10 @@ def make_image(state: MkosiState, skip: Sequence[str] = [], split: bool = False)
         definitions = state.workspace / "repart-definitions"
         if not definitions.exists():
             definitions.mkdir()
-            bootloader = state.root / f"efi/EFI/BOOT/BOOT{state.config.architecture.to_efi().upper()}.EFI"
+            if (a := state.config.architecture.to_efi()):
+                bootloader = state.root / f"efi/EFI/BOOT/BOOT{a.upper()}.EFI"
```
Comment (behrmann) [inline mkosi/__init__.py:1688]: Let's not be stingy and call it `arch` like above.
```suggestion
            if (arch := state.config.architecture.to_efi()):
                bootloader = state.root / f"efi/EFI/BOOT/BOOT{arch.upper()}.EFI"
```

File: mkosi/__init__.py [hunk 3]
```diff
@@ -1693,7 +1705,7 @@ def make_image(state: MkosiState, skip: Sequence[str] = [], split: bool = False)
                 )
 
             esp = (state.config.bootable == ConfigFeature.enabled or
-                  (state.config.bootable == ConfigFeature.auto and bootloader.exists()))
+                  (state.config.bootable == ConfigFeature.auto and bootloader and bootloader.exists()))
```
Comment (behrmann) [inline mkosi/__init__.py:1708]: Not introduced here, but the indentation is a bit confusing, either
```python
            esp = (state.config.bootable == ConfigFeature.enabled or
                   (state.config.bootable == ConfigFeature.auto and bootloader and bootloader.exists()))
```
or
```python
            esp = (
                state.config.bootable == ConfigFeature.enabled or
                (state.config.bootable == ConfigFeature.auto and bootloader and bootloader.exists())
            )
```
would be clearer.

File: mkosi/qemu.py [hunk 4]
```diff
@@ -219,19 +220,20 @@ def run_qemu(args: MkosiArgs, config: MkosiConfig) -> None:
...
         die(f"{config.output_format} images cannot be booted in qemu")
 
-    if config.output_format == OutputFormat.uki and config.qemu_firmware not in (QemuFirmware.auto, QemuFirmware.uefi):
-        die(f"uki images cannot be booted with the '{config.qemu_firmware}' firmware")
-
-    if config.output_format == OutputFormat.cpio and config.qemu_firmware not in (QemuFirmware.auto, QemuFirmware.linux, QemuFirmware.uefi):
-        die(f"cpio images cannot be booted with the '{config.qemu_firmware}' firmware")
+    if config.output_format in (OutputFormat.cpio, OutputFormat.uki) and\
+        config.qemu_firmware not in (QemuFirmware.auto, QemuFirmware.linux, QemuFirmware.uefi):
+        die(f"{config.output_format} images cannot be booted with the '{config.qemu_firmware}' firmware")
```
Comment (behrmann) [inline mkosi/qemu.py:225]: Line continuations are hard to spot, let's use parentheses
```suggestion
    if (config.output_format in (OutputFormat.cpio, OutputFormat.uki) and
        config.qemu_firmware not in (QemuFirmware.auto, QemuFirmware.linux, QemuFirmware.uefi)):
        die(f"{config.output_format} images cannot be booted with the '{config.qemu_firmware}' firmware")
```
This way one knows and needs to look at the end of the line too, to see where the condition ends.
```suggestion
    if (
        config.output_format in (OutputFormat.cpio, OutputFormat.uki) and
        config.qemu_firmware not in (QemuFirmware.auto, QemuFirmware.linux, QemuFirmware.uefi)
    ):
        die(f"{config.output_format} images cannot be booted with the '{config.qemu_firmware}' firmware")
```
would also visually separate the body.

**Review by behrmann — COMMENTED** [review]
LGTM functionality-wise, a few style nits.

**Review by behrmann — APPROVED** [review]

**Pull request merged**